Added a 'from_hsl' function to core rgba.cr

Demonstrandum opened the pull request:
The `from_hsl` function allows writing to an image through the HSL colour format, it converts the HSL value in to RGB values and uses a bit-depth of 8 using the preexisting `from_rgb_n(r, g, b, 8)` function.
Here's an example using the modified stumpy_core rgba.cr file:
```ruby
require "stumpy_png"
include StumpyPNG

spectrum = Canvas.new(361, 101)
(0..360.each do |x|
  (0..100).each do |y|
    # RGBA.from_hsla_n(hue, saturation, lightness, alpha, bit_depth) is an internal helper method
    color = RGBA.from_hsla_n(x, 100, y, UInt16::MAX, 8) # Here from_hsl(x, 100, y) would work exactly the same
    spectrum[x, y] = color
  end
end

StumpyPNG.write(spectrum, "spectrum.png")
```
Which produces:
![spectrum](https://cloud.githubusercontent.com/assets/26842759/26288203/ac984b64-3e84-11e7-9fe1-1ef252ffbf56.png)

`h` (hue) is in degrees (from 0°..360°) and
`s` (saturation) & `l` (lightness) are in percent, (0%..100%)

Commit:
New from_hsl function.

l3kn commented on src/stumpy_core/rgba.cr:
```diff
@@ -120,6 +120,32 @@ module StumpyCore
+        b = hue.call(a, b, h - 1.0/3.0)
+      end
+
+      return from_rgb_n((r * 255.0).round.to_i, (g * 255.0).round.to_i, (b * 255.0).round.to_i, 8)
```
You could use `from_relative(r, g, b, 1.0)` here

l3kn commented on src/stumpy_core/rgba.cr:
```diff
@@ -120,6 +120,32 @@ module StumpyCore
+      if s == 0
+        r = g = b = l
+      else
+        hue = Proc(Float64, Float64, Float64, Float64).new do | a, b, c |
```
Why not make this a private helper function like `get_hsl_hue`?

l3kn commented on src/stumpy_core/rgba.cr:
```diff
@@ -120,6 +120,32 @@ module StumpyCore
+      h /= 360.0
+      s /= 100.0
+      l /= 100.0
+
```
For `from_relative` to work, `r, g, b` must be floats
and I like initializing them unconditionally better

``` crystal
r = g = b = 1.0

if s != 0
...
```

l3kn reviewed (commented):
Thanks for the pull request!

I added some comments
to the parts of the code I'm not so sure about

There is some minor stuff to do
like increasing the version in `shard.yml` to `1.4.1`
and documenting the function,
but I can do that later, there is some other stuff missing
in the docs ;)

Commit:
Added more hsl funcs and worked on suggestions

Commit:
Increased version n˚

Commit:
HSL information and example

Commit:
hsl spectrum image

Commit:
from_hsla_n & from_hsl_n work with arrays/tuples

Commit:
Added from_hsla_n and from_hsl_n to interface

Demonstrandum commented:
Updated code for suggestions, added more HSL type functions (from_hsla_n, from_hsla, from_hsl_n, from_hsl, etc…) (also works with tuples/arrays), increased version and added information to README

l3kn commented:
Nice!

What would be the use case for `from_hsla_n` etc?
The `from_rgba_n` could be used when working with image formats
that enocde colors in 1bit / 2bit / 4bit / 8bit per channel,
but from what I've seen hsl uses either (0..360, 0..100, 0..100)
or (0..1, 0..1, 0..1) for the values.

Demonstrandum commented:
That's true! bit-depth would be completely unnecessary using HSL as it will always be between `0..360` and `0..100`. RGB values would always have the potential to be between 0..255 using this method. I'll remove the functions with options for bit-depth.

Commit:
Unnecessary bit-depth options removed!

Commit:
Fixed old hsla_n/hsl_n references
